pandas/tests/io/formats/test_style.py

```diff
@@ -1028,10 +1028,25 @@ def test_background_gradient(self):
             assert all("#" in x[0] for x in result.values())
             assert result[(0, 0)] == result[(0, 1)]
             assert result[(1, 0)] == result[(1, 1)]
+            for res in result:
+                if result[res][0].split(' ')[1] in ['#fde725', '#ffffcc']:
+                    assert result[(res)][1].split(' ')[1] == '#000000'
+                elif result[res][0].split(' ')[1] in ['#800026', '#440154']:
+                    assert result[(res)][1].split(' ')[1] == '#f1f1f1'
 
         result = df.style.background_gradient(
             subset=pd.IndexSlice[1, 'A'])._compute().ctx
-        assert result[(1, 0)] == ['background-color: #fff7fb']
+
+        assert result[(1, 0)] == ['background-color: #fff7fb',
+                                  'color: #000000']
+
+    @td.skip_if_no_mpl
+    def test_text_color_threshold(self):
+        df = pd.DataFrame([[1, 2], [2, 4]], columns=['A', 'B'])
+        for text_color_threshold in [1.1, '1', -1, [2, 2]]:
+            with pytest.raises(ValueError):
```

Comment:
`tm.assert_raises_regex` would be preferable as you can use it to assert on the message as well. Can still be used as a context manager (can also see usage in `test_to_html.py`)

Reply:
Thanks for the pointers, I updated accordingly.

This test still complains that a `ValueError` is not raised for any of the parameters. When I call the the function outside the test, it does raise `ValueError: ('`text_color_threshold` must be a value from 0 to 1.', 'occurred at index A')`. The test works fine if I raise `ValueError(msg)` manually.

Reply:
This didn't raise for me locally, though it did when running the `_compute` method

```python
In [12]: df.style.background_gradient(text_color_threshold='1')
Out[12]: <pandas.io.formats.style.Styler at 0x110163630>

In [13]: df.style.background_gradient(text_color_threshold='1')._compute()
ValueError: ('`text_color_threshold` must be a value from 0 to 1.', 'occurred at index A')
```

Reply:
Thanks, I missed that `_compute()` is called automatically in Notebooks via `_repr_html_`. I believe all comments are addressed now.